server/non_ui_views.py

```diff
@@ -250,17 +250,25 @@ def checkin(request):
         machine.activity = False
         machine.errors = machine.warnings = 0
         machine.save()
-        return HttpResponse(f"Sal report submmitted for {data.get('name', '')} with no activity")
+        return HttpResponse(f"Sal report submitted for {data.get('name', '')} with no activity")
 
     # If we get something back, we know the data is good, so store
-    # the bytes.
-    machine.report = report_bytes
+    # the bytes as unicode (otherwise it gets munged).
+    machine.report = report_bytes.decode()
     machine.console_user = get_console_user(report_data)
 
+    # We need to save now or else further processing of related fields
+    # will fail.
+    try:
+        machine.save()
+    except ValueError:
```

Comment:
If failing here means that nothing will be processed, should we bother continuing or just bail?

Reply:
Probably just bail. I didn't want it to throw an exception; but we should probably return an error then.